api/authentication.go

```diff
@@ -37,16 +37,19 @@ type Authenticator func(ctx context.Context, s *http.Server, r *http.Request) (e
 
 // SetAuthenticator sets an authenticator function for the API endpoint. If none is set, all requests will be permitted.
 func SetAuthenticator(fn Authenticator) error {
+	if module.Online() {
+		return ErrAuthenticationAlreadySet
```

Comment:
IMHO that error should be something like ErrModuleOnline because we don't know if the Authenticator is set.